aten/src/ATen/test/basic.cpp

```diff
@@ -1,290 +1,322 @@
-
-static void test(Type & type) {
-  CATCH_SECTION( "resize" ) {
-    auto a = at::empty({0}, type.options());
-    a.resize_({3,4});
-    CATCH_REQUIRE(a.numel() == 12);
-    a.resize_({5, 7});
-    CATCH_REQUIRE(a.numel() == 35);
-
+#define ASSERT_EQ_RESOLVED(X, Y) \
+  {                              \
+    bool isEQ = X == Y;          \
+    ASSERT_TRUE(isEQ);           \
   }
 
-  CATCH_SECTION( "ones and dot" ) {
-    Tensor b0 = ones({1, 1}, type);
-    CATCH_REQUIRE(2 == (b0+b0).sum().item<double>());
+using namespace at;
+
+void TestResize(Type& type) {
+  auto a = at::empty({0}, type.options());
+  a.resize_({3, 4});
+  ASSERT_EQ_RESOLVED(a.numel(), 12);
+  a.resize_({5, 7});
+  ASSERT_EQ_RESOLVED(a.numel(), 35);
+}
 
-    Tensor b1 = ones({1, 2}, type);
-    CATCH_REQUIRE(4 == (b1+b1).sum().item<double>());
+void TestOnesAndDot(Type& type) {
+  Tensor b0 = ones({1, 1}, type);
+  ASSERT_EQ_RESOLVED((b0 + b0).sum().item<double>(), 2);
 
-    Tensor b = ones({3, 4}, type);
-    CATCH_REQUIRE(24 == (b+b).sum().item<double>());
-    CATCH_REQUIRE(12 == b.numel());
-    CATCH_REQUIRE(b.view(-1).dot(b.view(-1)).item<double>() == 12);
-  }
+  Tensor b1 = ones({1, 2}, type);
+  ASSERT_EQ_RESOLVED((b1 + b1).sum().item<double>(), 4);
 
-  CATCH_SECTION( "rand" ) {
-    for(auto i = 0; i < 10; i++) {
-      Tensor a = rand({3,4}, type.toScalarType(i % 2 == 0 ? kFloat : kDouble));
-    }
-  }
+  Tensor b = ones({3, 4}, type);
+  ASSERT_EQ_RESOLVED((b + b).sum().item<double>(), 24);
+  ASSERT_EQ_RESOLVED(b.numel(), 12);
+  ASSERT_EQ_RESOLVED(b.view(-1).dot(b.view(-1)).item<double>(), 12);
+}
 
-  CATCH_SECTION( "sort" ) {
-    Tensor b = rand({3, 4}, type);
+void TestSort(Type& type) {
+  Tensor b = rand({3, 4}, type);
 
-    auto z = b.sort(1);
-    auto z_sorted = std::get<0>(z);
+  auto z = b.sort(1);
+  auto z_sorted = std::get<0>(z);
 
-    CATCH_REQUIRE(z_sorted[0][0].item<float>() < z_sorted[0][1].item<float>());
-  }
+  bool isLT = z_sorted[0][0].item<float>() < z_sorted[0][1].item<float>();
+  ASSERT_TRUE(isLT);
+}
 
-  if(type.backend() != Backend::CUDA)
-  CATCH_SECTION( "randperm" ) {
+void TestRandperm(Type& type) {
+  if (type.backend() != Backend::CUDA) {
     Tensor b = randperm(15, type);
     Tensor rv, ri;
     std::tie(rv, ri) = sort(b, 0);
-    CATCH_REQUIRE(rv[0].item<float>() <= rv[1].item<float>());
-  }
-
-  CATCH_SECTION( "context" ) {
-    std::stringstream ss;
-    ss << "context: " << std::hex << (int64_t)&globalContext() << std::endl;
+    bool isLE = (rv[0].item<float>() <= rv[1].item<float>());
+    ASSERT_TRUE(isLE);
   }
+}
 
-  CATCH_SECTION( "add" ) {
-    Tensor a = rand({3, 4}, type);
-    Tensor b = rand({3, 4}, type);
-    Tensor c = add(a, add(a, b));
-    //TODO:0-dim Tensor d(3.f);
-    Scalar d = 3.f;
-    CATCH_REQUIRE( add(c, d).allclose(a + a + b + d) );
-  }
+void SendContext() {
+  std::stringstream ss;
+  ss << "context: " << std::hex << (int64_t)&globalContext() << std::endl;
+}
```

Comment:
???

Reply:
This is from the original catch test:
CATCH_SECTION( "context" ) {
    std::stringstream ss;
    ss << "context: " << std::hex << (int64_t)&globalContext() << std::endl;
}

Although as you can see we dont know why it exist, I kept it unchanged for now.